File: frontend/src/services/productService.js
import api from '../api/axios';

export const getProducts = async () => {
  const { data } = await api.get('/products');
  return data;
};

export const createProduct = async (dto) => {
  const { data } = await api.post('/products', dto);
  return data;
};

export const updateProduct = async (id, dto) => {
  const { data } = await api.put(`/products/${id}`, dto);
  return data;
};

export const deleteProduct = async (id) => {
  const { status } = await api.delete(`/products/${id}`);
  return status;
};

File: backend/src/main/java/io/github/Lordrap2002/inventory/api/service/ProductService.java
package io.github.Lordrap2002.inventory.api.service;

import io.github.Lordrap2002.inventory.api.dto.ProductRequestDTO;
import io.github.Lordrap2002.inventory.api.dto.ProductResponseDTO;
import java.util.List;

public interface ProductService {
    ProductResponseDTO createProduct(ProductRequestDTO productRequestDTO);

    ProductResponseDTO updateProduct(Long id, ProductRequestDTO dto);

    void deleteProduct(Long id);

    List<ProductResponseDTO> getAllProducts();

    ProductResponseDTO getProductByID(Long id);
}
